pkgs/macro_service/lib/src/macro_service.dart

```diff
@@ -18,3 +18,20 @@ abstract interface class MacroService {
   /// Handles [request].
   Future<Response> handle(HostRequest request);
 }
+
+/// Shared implementation of auto incrementing 32 bit IDs.
+///
+/// These roll back to 0 once it is greater than 2^32.
+///
+/// These are only unique to the process which generates the request,
+/// so for instance the host and macro services may generate conflicting ids
+/// and that is allowed.
+int get nextRequestId {
+  final next = _nextRequestId++;
+  if (_nextRequestId > 2 ^ 32) {
```

Comment:
Ehhm ... 2 ^ 32 is 34 . Maybe 0x7fff ffff?

Reply:
lmao!